src/test/java/uk/gov/companieshouse/officer/delta/processor/service/api/ApiClientServiceImplTest.java

```diff
@@ -1,92 +1,221 @@
     }
 
     @Test
-    void shouldFailDeleteOfficerWhenMissingDeltaAt() {
-        ApiClientServiceImpl apiClientServiceSpy = Mockito.spy(apiClientService);
+    void shouldCallResponseHandlerWhenApiErrorResponseExceptionCaughtDuringDELETE() throws Exception {
+        // given
+        when(internalApiClientFactory.get()).thenReturn(client);
+        when(client.privateDeltaResourceHandler()).thenReturn(privateDeltaResourceHandler);
+        when(privateDeltaResourceHandler.deleteOfficer(anyString(), anyString(), anyString())).thenReturn(privateOfficerDelete);
+        when(privateOfficerDelete.execute()).thenThrow(ApiErrorResponseException.class);
+
+        DeleteAppointmentParameters deleteAppointmentParameters = DeleteAppointmentParameters.builder()
+                .encodedOfficerId(OFFICER_ID)
+                .deltaAt(DELTA_AT)
+                .encodedInternalId(APPOINTMENT_ID)
+                .companyNumber(COMPANY_NUMBER)
+                .build();
+
+        final String expectedUri = String.format("/company/%s/appointments/%s/full_record", COMPANY_NUMBER, APPOINTMENT_ID);
+
+        // when
+        apiClientService.deleteAppointment(deleteAppointmentParameters);
+
+        // then
+        verify(privateDeltaResourceHandler).deleteOfficer(expectedUri, DELTA_AT, OFFICER_ID);
+        verify(responseHandler).handle(any(ApiErrorResponseException.class));
+        verify(responseHandler, times(0)).handle(any(URIValidationException.class));
+    }
 
-        Executable executable = () ->  apiClientServiceSpy.deleteAppointment("context_id",
-                "N-YqKNwdT_HvetusfTJ0H0jAQbA", "09876543", null);
+    @Test
+    void shouldCallResponseHandlerWhenURIValidationExceptionCaughtDuringDELETE() throws Exception {
+        // given
+        when(internalApiClientFactory.get()).thenReturn(client);
+        when(client.privateDeltaResourceHandler()).thenReturn(privateDeltaResourceHandler);
+        when(privateDeltaResourceHandler.deleteOfficer(anyString(), anyString(), anyString())).thenReturn(privateOfficerDelete);
+        when(privateOfficerDelete.execute()).thenThrow(URIValidationException.class);
+
+        DeleteAppointmentParameters deleteAppointmentParameters = DeleteAppointmentParameters.builder()
+                .encodedOfficerId(OFFICER_ID)
+                .deltaAt(DELTA_AT)
+                .encodedInternalId(APPOINTMENT_ID)
+                .companyNumber(COMPANY_NUMBER)
+                .build();
+
+        final String expectedUri = String.format("/company/%s/appointments/%s/full_record", COMPANY_NUMBER, APPOINTMENT_ID);
+
+        // when
+        apiClientService.deleteAppointment(deleteAppointmentParameters);
+
+        // then
+        verify(privateDeltaResourceHandler).deleteOfficer(expectedUri, DELTA_AT, OFFICER_ID);
+        verify(responseHandler).handle(any(URIValidationException.class));
+        verify(responseHandler, times(0)).handle(any(ApiErrorResponseException.class));
```

Comment:
Nitpick but I would prefer to use `verifyNoInteractions()` and `verifyNoMoreInteractions()` or only use `verify(..., times())` instead of a mix of the 3. Personally I think the first two are simpler.